### PR TITLE
Set `javafmtOnCompile := false` by default

### PR DESCRIPTION
**Pull request by armanbilge**
It seems like this was already decided in https://github.com/sbt/sbt-java-formatter/pull/70#issuecomment-593421099 but was deferred to enable a smooth transition.

### CODE REVIEW

**Comment by ennru** [comment]
I agree, it is time to finally do this switch. Please update even the instructions in the Readme.

**Comment by armanbilge** [comment]
In aa2f583 I made an `on-compile-true` and `on-compile-false` test from the old `on-compile`.

File: plugin/src/sbt-test/sbt-java-formatter/on-compile-true/build.sbt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// no settings needed
```
Comment (ennru) [inline plugin/src/sbt-test/sbt-java-formatter/on-compile-true/build.sbt:1]:
```suggestion
```

**Review by ennru — APPROVED** [review]
LGTM.

**Pull request merged**